base/iostream.jl

```diff
@@ -321,13 +321,18 @@ function peek(s::IOStream)
     ccall(:ios_peekc, Cint, (Ptr{Void},), s)
 end
 
+
+function _seekback(pos::Integer, s::IOStream)
+    ccall(:ios_seekback, Cint, (Cint, Ptr{Void}), pos, s)
+end
+
 function skipchars(io::IOStream, pred; linecomment=nothing)
     while !eof(io)
         c = read(io, Char)
```

Comment:
That's what the older implementation did, but then you have to do twice the work for every character, so it was slower. I figured a single backtrack at the end of the function would be better than a backtrack for every skipped character.